### DISCUSSION TITLE
ValueError: source code string cannot contain null bytes

### DISCUSSION THREAD

**Discussion by music1999** [fishaudio/fish-speech#603] [Q&A]
I installed it through Python and downloaded the inference model. However, when executing the inference, it prompted that the tools module could not be found. I used pip install tools to install it, but it seems to be incorrect. The error still occurs during execution
Traceback (most recent call last):
File "D:\openai.wiki\fish-speech-main\tools\vqgan\inference.py",  line 14, in <module>
from tools.file import AUDIO_EXTENSIONS
ValueError: source code string cannot contain null bytes
Is it because I installed the wrong version of tools or the wrong library file name?

**Comment by AnyaCoder** [comment]
Type this command :
 ```bash
cd /d D:\openai.wiki\fish-speech-main
set PYTHONPATH=D:\openai.wiki\fish-speech-main
```

**Comment by music1999** [comment]
(fishspeech_env) D:\openai.wiki\fish-speech-main>python tools/vqgan/inference.py -i C:\Users\Administrator\Desktop\wav\Track_5.wav --checkpoint-path "checkpoints/fish-speech-1.4/firefly-gan-vq-fsq-8x1024-21hz-generator.pth"
Traceback (most recent call last):
  File "D:\openai.wiki\fish-speech-main\tools\vqgan\inference.py", line 14, in <module>
    from tools.file import AUDIO_EXTENSIONS
ValueError: source code string cannot contain null bytes

This is the entire error message, and I have already set the Python environment variables。

Here is the information to verify the successful setting of Python environment variables

(fishspeech_env) D:\openai.wiki\fish-speech-main>python
Python 3.10.9 | packaged by Anaconda, Inc. | (main, Mar  1 2023, 18:18:15) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>

There is no problem with the settings, I think it may be due to installation errors of other libraries? Is there a file called 'requirements' that guides the installation of the library?

**Reply by AnyaCoder** [reply]
You can try to follow docs for `non-professional windows user` for environment installation.
